### DISCUSSION TITLE
Default profile picture?

### DISCUSSION THREAD

**Discussion by Nv7-GitHub** [pocketbase/pocketbase#1627] [Q&A] [answered]
When the `avatar` field in a user is N/A, is it possible for all requests for that file to instead return a "default" image (in this case, one of those blank user icons)? Do I have to implement this client-side instead?

**Comment by ganigeorgiev** [comment, marked as answer]
There are no "defaults" unless you define it client-side or server-side in a [before hook](https://pocketbase.io/docs/event-hooks/).

**Reply by Nv7-GitHub** [reply]
I see, I will try using the `OnFileDownloadRequest` hook to do this. Thank you!

**Comment by dschissler** [comment]
@Nv7-GitHub Here is some code for that.  https://github.com/pocketbase/pocketbase/discussions/1418

I put the hooks for the user collection in it's own package, but you could inline it if it doesn't bother you.  I helped put in the `filesystem.NewFileFromBytes` function, which without it is cumbersome because the file needs to exist first.  If the file already exists locally then you can use `filesystem.NewFileFromPath`.

Also if you were giving the users the option to set the avatar during the account creation then you would want to check if the file exists already.

Also the `form.SetDao(daos.New(env.Dao().DB()))` line is necessary to prevent an infinite loop or another unwanted update event.  It creates new dao (ORM) that does not have the app events registered.  So you won't get realtime events from this, and it won't trigger the model events.  You could also solve this using some strange flags, but that is unnecessary when there is a more straightforward solution.

```golang
	app.OnModelAfterCreate().Add(func(e *core.ModelEvent) error {
		m, ok := e.Model.(*pbmodels.Record)
		if !ok {
			return nil
		}

		switch m.TableName() {
		case models.UsersCollectionName:
			return user_hooks.AfterCreate(env, m)
		}

		return nil
	})
```

```golang
func AfterCreate(env *svc.Env, m *pbmodels.Record) error {
	if m.GetString("avatar") == "" {
		return generateAvatar(env, m)
	}

	return nil
}

func generateAvatar(env *svc.Env, m *pbmodels.Record) error {
	g := govatar.MALE

	b, err := avatar.GenerateForName(g, m.GetString("username"), models.UsersAvatarSize)
	if err != nil {
		return err
	}

	f, err := filesystem.NewFileFromBytes(b, "avatar.jpg")
	if err != nil {
		return errors.New("unable to create avatar file from bytes")
	}

	form := forms.NewRecordUpsert(env.App, m)
	form.SetDao(daos.New(env.Dao().DB()))
	form.AddFiles("avatar", f)

	return form.Submit()
}
```
